src/tools/rbd/action/Bench.cc

```diff
@@ -117,15 +120,15 @@ struct rbd_bencher {
       io_type(io_type),
       io_size(io_size)
   {
-    if (io_type == IO_TYPE_WRITE) {
+    if (io_type == IO_TYPE_WRITE || io_type == IO_TYPE_RW) {
       bufferptr bp(io_size);
       memset(bp.c_str(), rand() & 0xff, io_size);
       write_bl.push_back(bp);
     }
   }
 
 
-  bool start_io(int max, uint64_t off, uint64_t len, int op_flags)
+  bool start_io(int max, uint64_t off, uint64_t len, int op_flags, bool read_flag)
```

Comment:
Use ```read_flag``` to determine if it's a read vs write operation -- i.e. replace the current ```io_type``` switch w/ ```if``` / ```else``` block.